tools/buildsys/src/builder.rs

```diff
@@ -86,269 +85,355 @@ lazy_static! {
+            ),
+            target_build_args: TargetBuildArgs::Variant(VariantBuildArgs {
+                data_image_publish_size_gib,
+                data_image_size_gib: data_image_size_gib.to_string(),
+                image_features: manifest
+                    .image_features()
+                    .unwrap_or_default()
+                    .into_iter()
+                    .copied()
+                    .collect(),
+                image_format: match manifest.image_format() {
+                    Some(ImageFormat::Raw) | None => "raw",
+                    Some(ImageFormat::Qcow2) => "qcow2",
+                    Some(ImageFormat::Vmdk) => "vmdk",
+                }
+                .to_string(),
+                kernel_parameters: manifest
+                    .kernel_parameters()
+                    .cloned()
+                    .unwrap_or_default()
+                    .join(" "),
+                name: args.name,
+                os_image_publish_size_gib: os_image_publish_size_gib.to_string(),
+                os_image_size_gib: os_image_size_gib.to_string(),
+                packages: manifest
+                    .included_packages()
+                    .cloned()
+                    .unwrap_or_default()
+                    .join(" "),
+                partition_plan: match partition_plan {
+                    PartitionPlan::Split => "split",
+                    PartitionPlan::Unified => "unified",
+                }
+                .to_string(),
+                pretty_name: args.pretty_name,
+                variant: args.variant,
+                version_build: args.version_build,
+                version_image: args.version_image,
+            }),
+            secrets_args: secrets_args()?,
+        })
+    }
 
-    // Clean up any previous outputs we have tracked.
-    clean_build_files(&build_dir, output_dir)?;
+    pub(crate) fn build(&self) -> Result<()> {
+        env::set_current_dir(&self.root_dir).context(error::DirectoryChangeSnafu {
+            path: &self.root_dir,
+        })?;
 
-    let target = match kind {
-        BuildType::Package => "package",
-        BuildType::Variant => "variant",
-    };
+        // Create a directory for tracking outputs before we move them into position.
+        let marker_dir = create_marker_dir(
+            &self.target_build_args.build_type(),
+            &self.artifact_name,
+            &self.common_build_args.arch.to_string(),
+            &self.state_dir,
+        )?;
 
-    // Our dockerfile is in the Twoliter tools directory.
-    let twoliter_tools_dir =
-        env::var(TOOLS_DIR).context(error::EnvironmentSnafu { var: TOOLS_DIR })?;
-    let dockerfile = PathBuf::from(twoliter_tools_dir).join("Dockerfile");
+        // Clean up any previous outputs we have tracked.
+        clean_build_files(&marker_dir, &self.artifacts_dir)?;
 
-    let mut build = format!(
-        "build . \
+        let mut build = format!(
+            "build {context} \
```

Comment:
nit: this intending looks strange to me?

Reply:
> nit: this intending looks strange to me?

Yeah fmt is no help here. I've now aligned the multiline string line starts with the variable list below it.